src/Microsoft.IdentityModel.Xml/X509Data.cs

```diff
@@ -0,0 +1,150 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.IdentityModel.Xml
+{
+    /// <summary>
+    /// Represents a XmlDsig X509Data element as per:  https://www.w3.org/TR/2001/PR-xmldsig-core-20010820/#sec-X509Data
+    /// </summary>
+    /// <remarks> Supports multiple certificates. </remarks>
+    public class X509Data
+    {
+        /// <summary>
+        /// Initializes an instance of <see cref="X509Data"/>.
+        /// </summary>
+        public X509Data()
+        {
+        }
+
+        /// <summary>
+        /// Initializes an instance of <see cref="X509Data"/>.
+        /// </summary>
+        public X509Data(X509Certificate2 certificate)
+        {
+            if (certificate != null)
+                Certificates.Add(Convert.ToBase64String(certificate.RawData));
+        }
+
+
+        /// <summary>
+        /// Initializes an instance of <see cref="X509Data"/>.
+        /// </summary>
+        public X509Data(IEnumerable<X509Certificate2> certificates)
+        {
+            if (certificates != null)
+            {
+                foreach (var certificate in certificates)
+                {
+                    if (certificate != null)
+                        Certificates.Add(Convert.ToBase64String(certificate.RawData));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the 'X509IssuerSerial' that is part of a 'X509Data'.
+        /// </summary>
+        public IssuerSerial IssuerSerial
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Gets or sets the 'X509SKI' value that is a part of 'X509Data'.
+        /// </summary>
+        public string SKI
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Get or sets the 'X509SubjectName' value that is a part of 'X509Data'.
+        /// </summary>
+        public string SubjectName
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Get the collection of X509Certificates that is associated with 'X509Data'.
+        /// </summary>
+        public ICollection<string> Certificates { get; } = new Collection<string>();
+
+        /// <summary>
+        /// Get or sets the 'CRL' value that is a part of 'X509Data'.
+        /// </summary>
+        public string CRL
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Checks if an 'X509Data' object is empty.
+        /// </summary>
+        public bool IsEmpty()
```

Comment:
I don't think we need this. When reading, throw if the X509Data is an EmptyElement

Reply:
That makes sense!